### ISSUE TITLE
memory leak

### ISSUE THREAD

**Issue by theAeon** [PKBeam/AMWin-RP#40]
![image](https://github.com/PKBeam/AMWin-RP/assets/9099529/a6107a8a-ac67-4ed3-a5a2-72dee907ce22)

Hi! Appreciate the work done on this but whenever I leave it open in the background for long enough it ends up eating as much RAM as it can get its hand on. Unsure how to test further.

**Comment by theAeon** [comment]
Currently pulling a minidump - outdated, granted, as I didn't know about the 1.2.5 release.

**Comment by theAeon** [comment]
https://robbinsa.me/files/AMWin-RichPresence.exe_2023-10-04_10-12-33.dmp.zst

**Comment by PKBeam** [comment]
I took some memory snapshots and for me the application appeared to leak about 60 MB in 10 minutes.

Might be related to the HTTP requests? There are also Newtonsoft Json objects showing in the heap - so possibly something to do with RPC or scrobbling. Not really sure where the offending code is though.

**Comment by th0mk** [comment]
When I was messing around with it locally I noticed it keeps refreshing through `GetAppleMusicInfo` every few seconds, even if the track has not changed. I can imagine this can cause it to leak memory, since there are at least two methods in there which scrape the Apple Music website for info about the track.

**Comment by theficus** [comment]
I noticed the same memory leak issue. After a few days, it was using well over 5GB of memory. Pretty unreasonable for such a simple app.

I looked at the code and I believe I found the likely culprit, and that's the creation of HttpClient instances sprinkled throughout the code. There's a couple problems here; these are IDisposable and they aren't being disposed anywhere. But even if they were being disposed, Microsoft's documentation recommends using [a single static instance for the lifetime of the process](https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines#recommended-use). 

I did a small refactor (https://github.com/theficus/AMWin-RP/commit/522eb8ab247ca3e6828c2e486cbb0fac227a4714) in an attempt to fix this issue and it seems to have made a big difference. When profiling the "before" code, I can see the memory continually going up after every HttpClient instance is created. When trying my fix, it stays pretty stable although I've not run it for an extended period of time. It shouldn't cause any harm, though.

I'll submit a PR for you to evaluate further.
